src/marker.rs

```diff
@@ -77,6 +77,63 @@ impl MarkerListener {
         let loop_lines = blines.clone();
         let local_listener = tf_listener.clone();
 
+        let _sub = rosrust::subscribe(
+            &config.topic,
+            2,
+            move |msg: rosrust_msg::visualization_msgs::Marker| {
+                let mut _lines = loop_lines.write().unwrap();
+                _lines.clear();
```

Comment:
Is this how you initialize.alloc memory a vector of this length?
why not use:
` Vec::with_capacity(size);`

Reply:
the length is not known depending on the marker type ( not that we implemented alot) i think here it does not make much sense as n is small